app/src/main/java/io/livekit/android/example/voiceassistant/TokenExt.kt

```diff
@@ -0,0 +1,65 @@
+package io.livekit.android.example.voiceassistant
+
+import android.util.Log
+import android.widget.Toast
+import androidx.activity.ComponentActivity
+import com.google.gson.Gson
+import okhttp3.*
+import java.io.IOException
+
+// TODO: Add your Sandbox ID here
+// When building an app for production, you should use your own token server.
+const val sandboxID = ""
+
+data class ConnectionDetails(
+    val serverUrl: String,
+    val participantToken: String,
+)
+
+/**
+ * Ensures a valid LiveKit token is generated
+ */
+fun ComponentActivity.requireToken(onTokenGenerated: (cd: ConnectionDetails) -> Unit) {
+    val activity = this
```

Comment:
is it worth having a stub in here that says `if sandBoxID.length == 0 { onTokenGenerated(url: "ADD_URL", token: "ADD_TOKEN"); return }` with a big comment about how to hardcode them instead?

Reply:
Yeah, makes sense!